test/framework/src/test/java/org/elasticsearch/test/disruption/NetworkDisruptionIT.java

```diff
@@ -102,4 +112,56 @@ public void testNetworkPartitionRemovalRestoresConnections() throws Exception {
         }
     }
 
+    public void testTransportRespondsEventually() throws InterruptedException {
+        internalCluster().setBootstrapMasterNodeIndex(0);
+        internalCluster().ensureAtLeastNumDataNodes(randomIntBetween(3, 5));
+        final NetworkDisruption.DisruptedLinks disruptedLinks;
+        if (randomBoolean()) {
+            disruptedLinks = TwoPartitions.random(random(), internalCluster().getNodeNames());
+        } else {
+            disruptedLinks = NetworkDisruption.Bridge.random(random(), internalCluster().getNodeNames());
+        }
+
+        NetworkDisruption networkDisruption = new NetworkDisruption(disruptedLinks, randomFrom(new NetworkDisruption.NetworkUnresponsive(),
+            new NetworkDisruption.NetworkDisconnect(), NetworkDisruption.NetworkDelay.random(random())));
+        internalCluster().setDisruptionScheme(networkDisruption);
+
+        networkDisruption.startDisrupting();
+
+        CountDownLatch latch = new CountDownLatch(100);
+        for (int i = 0; i < 100; ++i) {
+            TransportService transportService = internalCluster().getInstance(TransportService.class);
+            TransportService target = internalCluster().getInstance(TransportService.class);
+
+            transportService.sendRequest(target.getLocalNode(), ClusterHealthAction.NAME, new ClusterHealthRequest(),
+                new TransportResponseHandler<>() {
+                @Override
+                public void handleResponse(TransportResponse response) {
+                    latch.countDown();
+                }
+
+                @Override
+                public void handleException(TransportException exp) {
+                    latch.countDown();
+                }
+
+                @Override
+                public String executor() {
+                    return ThreadPool.Names.SAME;
+                }
+
+                @Override
+                public TransportResponse read(StreamInput in) throws IOException {
+                    return ClusterHealthResponse.readResponseFrom(in);
+                }
+            });
+        }
+
+        // give a bit of time to send something under disruption.
+        latch.await(500, TimeUnit.MILLISECONDS);
+        networkDisruption.stopDisrupting();
+
+        latch.await(30, TimeUnit.SECONDS);
```

Comment:
I think we should assert that this doesn't time out:

```suggestion
        assertTrue("all requests completed", latch.await(30, TimeUnit.SECONDS));
```

Reply:
I did this the other way around to ensure that a failure will include the count. In case this ever fails, it might be interesting to know if it was 1 request remaining or 100?

Let me know if you would still like me to change it into your suggestion?

Reply:
Ok, that makes sense.